stackslib/src/net/http/request.rs

```diff
@@ -425,8 +434,19 @@ impl StacksMessageCodec for HttpRequestPreamble {
                                 "Inavlid HTTP request: invalid Connection: header".to_string(),
                             ));
                         }
+                        seen_headers.insert(key);
+                    } else if key == "set-cookie" {
+                        set_cookie.push(value);
                     } else {
-                        headers.insert(key, value);
+                        match headers.entry(key) {
+                            Entry::Vacant(vacant_entry) => {
+                                vacant_entry.insert(value);
+                            }
+                            Entry::Occupied(mut occupied_entry) => {
+                                occupied_entry.get_mut().push_str(", ");
+                                occupied_entry.get_mut().push_str(&value);
+                            }
+                        }
```

Comment:
I think this could be:

```rust
                        headers
                            .entry(key)
                            .and_modify(|entry| {
                                entry.push_str(", ");
                                entry.push_str(&value);
                            })
                            .or_insert(value);
```